### PR TITLE
[JIT Kernel] Migrate kimi_k2_moe_fused_gate to JIT

### PR DESCRIPTION
**Pull request by xingsy97**
## Motivation

Migrate `kimi_k2_moe_fused_gate` kernel to JIT compilation (#17865).

## Modifications

Under `python/sglang/jit_kernel/`:
- `csrc/moe/kimi_k2_moe_fused_gate.cuh` — CUDA kernel (ported from `sgl-kernel/csrc/moe/kimi_k2_moe_fused_gate.cu`, both small_token and large_token variants)
- `kimi_k2_moe_fused_gate.py` — Python wrapper
- `tests/test_kimi_k2_moe_fused_gate.py` — Unit tests (JIT vs PyTorch)
- `benchmark/bench_kimi_k2_moe_fused_gate.py` — Benchmark (JIT vs AOT)

And
- `python/sglang/srt/layers/moe/topk.py` — Switch import to JIT

## Accuracy Tests

Pass all tests defined in `python/sglang/jit_kernel/tests/test_kimi_k2_moe_fused_gate.py`

## Benchmarking and Profiling

```
GPU: NVIDIA GeForce RTX 5060 Laptop GPU
Driver: 580.126.20 | CUDA: 12.8 | PyTorch: 2.9.1+cu128
Script: python/sglang/jit_kernel/benchmark/bench_kimi_k2_moe_fused_gate.py

kimi-k2-moe-fused-gate-performance (unit: us):
   M × topk         JIT      AOT      AOT->JIT
   1 × 2            1.70     1.70     0.0%
   8 × 6            3.15     3.11     +1.3%
   32 × 4           2.79     2.74     +1.8%
   128 × 6          8.83     8.84     -0.1%
   256 × 2          7.07     6.89     +2.6%
   512 × 6         27.04    27.67     -2.3%
   1024 × 2         9.11     9.10     +0.1%
   1024 × 6        14.60    14.82     -1.5%
```

Both small_token (m <= 512) and large_token (m > 512) kernel variants are ported; no regression observed.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

### CODE REVIEW

File: python/sglang/jit_kernel/tests/test_kimi_k2_moe_fused_gate.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    output_scores = torch.gather(sigmoid_vals, 1, top_indices)
+
+    if renormalize:
+        score_sum = output_scores.sum(dim=-1, keepdim=True)
+        output_scores = torch.where(
+            score_sum > 0,
+            output_scores / score_sum,
+            output_scores,
+        )
```
[the remark this answers is not in the document]
Comment (xingsy97) [reply python/sglang/jit_kernel/tests/test_kimi_k2_moe_fused_gate.py]: yeah you're right, torch.where can still evaluate the division even when the condition is false. switched to clamp(min=eps), much simpler.

**Comment by hnyls2002** [comment]
Thanks @xingsy97! This migration has already landed - `kimi_k2_moe_fused_gate` runs through JIT on main via `python/sglang/kernels/jit/csrc/trtllm_lora_temp/kimi_k2_moe_fused_gate.cuh` and `python/sglang/kernels/ops/moe/trtllm_lora_temp/kimi_k2_moe_fused_gate.py` (merged in #27329), and `python/sglang/srt/layers/moe/topk.py:1624` already imports the JIT wrapper. The `python/sglang/jit_kernel/` package this PR adds to was also removed by #31666 when JIT infra moved to `sglang.kernels`. Closing as superseded - please reopen if I've missed something.
